src/app/phase1/issue/issue.component.ts

```diff
@@ -1,67 +1,35 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { Issue } from '../../core/models/issue.model';
-import { IssueService } from '../../core/services/issue.service';
-import { FormBuilder } from '@angular/forms';
-import { finalize } from 'rxjs/operators';
-import { ErrorHandlingService } from '../../core/services/error-handling.service';
-import { IssueCommentService } from '../../core/services/issue-comment.service';
-import { Subscription } from 'rxjs';
+import { ViewIssueComponent, ISSUE_COMPONENTS } from '../../shared/view-issue/view-issue.component';
 
 @Component({
   selector: 'app-issue',
   templateUrl: './issue.component.html',
   styleUrls: ['./issue.component.css']
 })
-export class IssueComponent implements OnInit, OnDestroy {
-  issue: Issue;
-  isIssueLoading = true;
-  isEditing = false;
-  issueSubscription: Subscription;
+export class IssueComponent implements OnInit {
+  issueId: number;
 
-  constructor(private issueService: IssueService,
-              private issueCommentService: IssueCommentService,
-              private route: ActivatedRoute,
-              private formBuilder: FormBuilder,
-              private errorHandlingService: ErrorHandlingService) { }
+  readonly issue_components: ISSUE_COMPONENTS[] = [
+    ISSUE_COMPONENTS.TESTER_POST,
+    ISSUE_COMPONENTS.SEVERITY_LABEL,
+    ISSUE_COMPONENTS.TYPE_LABEL
+  ];
 
-  ngOnInit() {
-    this.initializeIssue();
-  }
-
-  canDeactivate() {
-    return !this.isEditing;
-  }
-
-  updateIssue(newIssue: Issue) {
-    this.issue = newIssue;
-    this.issueService.updateLocalStore(this.issue);
-  }
-
-  updateEditState(newState: boolean) {
-    this.isEditing = newState;
-  }
+  @ViewChild(ViewIssueComponent) view_issue: ViewIssueComponent;
```

Comment:
Any reason for using snake case instead of camel case?

Reply:
@JunWei96 ViewIssueComponent is a class type

Reply:
@JunWei96  I have made the correct changes

Reply:
`issue_component` is also in snake case.

Reply:
thanks for spotting that, I have changed it to the correct format